sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/TargetDialog.java

```diff
@@ -10,192 +9,189 @@
+          else if (t.trim().equals(""))
+              throw new ParseException("bad tag", 0);
+          this.setParameter(ParameterNames.LOCAL_TAG, t);
+    }
+    /** Boolean function
+     * @return true if the Tag exist
+     */
+    public boolean hasLocalTag() {
+        return hasParameter(ParameterNames.LOCAL_TAG);
+    }
+
+    /** remove Tag member
+     */
+    public void removeLocalTag() {
+        parameters.delete(ParameterNames.LOCAL_TAG);
+        
+    }
+    /**
+     * Gets the remote tag of the Target-Dialog header.
+     *
+     * @return the remote tag value
+     */
+    public String getRemoteTag() {
+    	 if (parameters == null)
+             return null;
+         return getParameter(ParameterNames.REMOTE_TAG);
+    }
+    /**
+     * Sets the remote tag of the Target-Dialog header.
+     *
+     * @param remoteTag the remote tag value to set
+     * @throws ParseException if the provided remoteTag is invalid
+     */
+    public void setRemoteTag(String t) throws ParseException {
+    	  if (t == null)
+              throw new NullPointerException("null tag ");
+          else if (t.trim().equals(""))
+              throw new ParseException("bad tag", 0);
+          this.setParameter(ParameterNames.REMOTE_TAG, t);
+    }
+    /** Boolean function
+     * @return true if the Tag exist
+     */
+    public boolean hasRemoteTag() {
+        return hasParameter(ParameterNames.REMOTE_TAG);
+    }
+
+    /** remove Tag member
+     */
+    public void removeRemoteTag() {
+        parameters.delete(ParameterNames.REMOTE_TAG);      
+    }
+
+    @Override
+    public void setValue(String value) throws ParseException {
+        throw new ParseException(value,0);
+    }
+
+
+    /**
+     * Parses the header string into Call-Id, local tag, remote tag, and parameters.
+     *
+     * @param body the header string to parse
+     * @throws ParseException if the header string cannot be parsed
+     */
+    public void decodeBody(String body) {
+        try {
+            // Split the header by semicolons to extract each parameter
+            String[] params = body.split(";");
+
+            // Extract the call ID from the first parameter
+            String callIdParam = params[0].trim();
+            int equalIndex = callIdParam.indexOf('=');
+            if (equalIndex != -1) {
+                setCallId(callIdParam.substring(equalIndex + 1).trim());
+            } else {
+                throw new ParseException("Invalid Target-Dialog header format: missing Call-Id", 0);
+            }
+
+            // Extract local and remote tags from subsequent parameters
+            for (int i = 1; i < params.length; i++) {
+                String param = params[i].trim();
+                equalIndex = param.indexOf('=');
+                if (equalIndex != -1) {
+                    String paramName = param.substring(0, equalIndex).trim();
+                    if (paramName.equals("local-tag")) {
```

Comment:
Let's use the constants you defined above in sip-ri/src/main/java/gov/nist/javax/sip/header/ParameterNames.java for local-tag and remote-tag

Reply:
done